File: src/lib/fix_alignment.js
const fs = require('fs');
const path = require('path');

function fixFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf8');
    let original = content;

    // 1. textAlign: (isRtl ? 'right' : 'left') or isRtl ? 'right' : 'left' -> textAlign: 'start'
    content = content.replace(/textAlign:\s*\(?\s*isRtl\s*\?\s*'right'\s*:\s*'left'\s*\)?/g, "textAlign: 'start'");
    
    // 2. textAlign: (isRtl ? 'left' : 'right') or isRtl ? 'left' : 'right' -> textAlign: 'end'
    content = content.replace(/textAlign:\s*\(?\s*isRtl\s*\?\s*'left'\s*:\s*'right'\s*\)?/g, "textAlign: 'end'");

    // 3. [isRtl ? 'right' : 'left']: -> insetInlineStart:
    content = content.replace(/\[\s*isRtl\s*\?\s*'right'\s*:\s*'left'\s*\]:/g, "insetInlineStart:");
    
    // 4. [isRtl ? 'left' : 'right']: -> insetInlineEnd:
    content = content.replace(/\[\s*isRtl\s*\?\s*'left'\s*:\s*'right'\s*\]:/g, "insetInlineEnd:");

    // 5. direction: 'rtl' -> direction: 'inherit' (to allow LTR switch)
    content = content.replace(/direction:\s*'rtl'/g, "direction: 'inherit'");

    // 6. !isRtl ? 'left' : 'right' -> 'start'
    content = content.replace(/textAlign:\s*\(?\s*!isRtl\s*\?\s*'left'\s*:\s*'right'\s*\)?/g, "textAlign: 'start'");

    if (content !== original) {
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`Fixed: ${filepath}`);
    }
}

function walk(dir) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (file !== 'node_modules' && file !== '.next' && file !== '.git') {
                walk(fullPath);
            }
        } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
            fixFile(fullPath);
        }
    });
}

const baseDir = path.join(__dirname, '..');
walk(baseDir);
console.log('Done.');

File: src/lib/fix_alignment.py
import os
import re

def fix_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # 1. textAlign isRtl ? 'right' : 'left' -> textAlign: 'start'
    new_content = re.sub(r"textAlign:\s*isRtl\s*\?\s*'right'\s*:\s*'left'", "textAlign: 'start'", content)
    
    # 2. textAlign isRtl ? 'left' : 'right' -> textAlign: 'end'
    new_content = re.sub(r"textAlign:\s*isRtl\s*\?\s*'left'\s*:\s*'right'", "textAlign: 'end'", new_content)

    # 3. [isRtl ? 'right' : 'left']: -> insetInlineStart: (usually used for absolute positioning)
    # This might be tricky, let's look at examples first.
    # From profile/page.tsx: [isRtl ? 'right' : 'left']: '14px'
    # This is equivalent to insetInlineStart: '14px'
    new_content = re.sub(r"\[isRtl\s*\?\s*'right'\s*:\s*'left'\]:", "insetInlineStart:", new_content)
    
    # 4. Same for 'left' : 'right' -> insetInlineEnd
    new_content = re.sub(r"\[isRtl\s*\?\s*'left'\s*:\s*'right'\]:", "insetInlineEnd:", new_content)

    if content != new_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Fixed: {filepath}")

def main():
    base_dir = r"c:\Users\pc203\OneDrive\Desktop\ERP\erp-app\src"
    for root, dirs, files in os.walk(base_dir):
        for file in files:
            if file.endswith(('.tsx', '.ts')):
                fix_file(os.path.join(root, file))

if __name__ == "__main__":
    main()
